nifi-extension-bundles/nifi-standard-bundle/nifi-standard-rules/src/main/java/org/apache/nifi/flowanalysis/rules/RequireSecureConnection.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.flowanalysis.rules;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.documentation.UseCase;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.ControllerService;
+import org.apache.nifi.flow.VersionedComponent;
+import org.apache.nifi.flow.VersionedExtensionComponent;
+import org.apache.nifi.flowanalysis.AbstractFlowAnalysisRule;
+import org.apache.nifi.flowanalysis.ComponentAnalysisResult;
+import org.apache.nifi.flowanalysis.FlowAnalysisRuleContext;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.ssl.SSLContextProvider;
+import org.apache.nifi.util.StringUtils;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+
+@Tags({"component", "processor", "controller service", "type"})
+@CapabilityDescription("Produces rule violations for each component (i.e. processors or controller services) having a property "
+        + "identifying an SSLContextService that is not set.")
+@UseCase(
+        description = "Ensure that no ports can be opened for insecure (plaintext, e.g.) communications.",
```

Comment:
Related to the name, this is not completely accurate for all cases.

Reply:
I'm open to suggestions for the name, the idea is that you can't add a ListenHttp to the canvas and open an insecure port.

Reply:
Thanks for the reply. If that's the case, it sounds like it would be cleaner to limit this rule to the ListenHTTP Processor. That would avoid a lot of the potential ambiguity.

Reply:
Maybe for all TCP-based Listen processors? ListenHTTP, ListenTCP, ListenOTLP, etc.?

Reply:
The Component Types property is a bit ambiguous as it stands, because "Component Type" could imply Processor versus Controller Service. `Component Classes` might be clearer, requiring the full class name. However, if the goal is to focus on components that can receive connections, it seems like the list could be embedded in the Rule itself, not requiring the property.

Reply:
I had to go a different route so I used a list of affected processors, but I also couldn't get at the actual PropertyDescriptor for the components, just the VersionedPropertyDescriptor, which does not identify the Controller Service reference, just that the property refers to a CS. This solution is brittle (based on property names) but I don't see a better way unless we can guarantee all components are InstantiatedVersionedComponents.